griptape/memory/task/storage/text_artifact_storage.py

```diff
@@ -36,12 +42,22 @@ def summarize(self, namespace: str) -> TextArtifact:
 
         return self.summary_engine.summarize_artifacts(self.load_artifacts(namespace))
 
-    def query(self, namespace: str, query: str, metadata: Any = None) -> TextArtifact | InfoArtifact:
+    def query(self, namespace: str, query: str, metadata: Any = None) -> BaseArtifact:
         if self.rag_engine is None:
             raise ValueError("RAG engine is not set.")
 
         result = self.rag_engine.process(
-            RagContext(initial_query=query, namespace=namespace, metadata=None if metadata is None else str(metadata))
+            RagContext(
+                query=query,
+                module_params={
+                    self.retrieval_rag_module_name: {  # pyright: ignore
```

Comment:
What is the pyright issue? If it's required, only ignore the specific error, not all errors.